site/docs/design-docs/013-system-time-watermarks.md

```diff
@@ -0,0 +1,50 @@
+---
+title: 013 - Resolving Sparse Events Issue with System-Time Watermarks
+description: Make the time progress independently from the event rate
+---
+
+*Target Release*: 4.3
+
+## The _Sparse Events Issue_
+
+When using event time, time progresses only through the ingestion of new
+events. If the events are sparse, time will effectively stop until a
+newer event arrives. This causes high latency for time-sensitive
+operations such as window aggregation. In addition, Jet tracks event
+time for every source partition separately, and if just one partition
+has sparse events, time progress in the whole job is hindered.
+
+## Ingestion Time
+
+When you use `withIngestionTime()` on the pipeline, Jet won't extract
+the timestamp from the event, but will set it according to the local
+system time on the server. This way of assigning timestamps is inferior
+because the event isn't associated with the time at which it occurred,
+but at which Jet received it, causing inaccuracy in the calculations. A
+specifically bad problem is that, if the job restarts due to a crash,
+Jet must receive again some of the events it already received and
+processed before the crash. These events will now get different
+timestamps, affecting a different piece of the aggregation state.
+
+On the other hand, this timestamping approach provides the best latency:
+the job will never wait for delayed items, there's no need to allow any
+event lag as with event-time processing.
+
+## The Improvement
+
+We first implemented `WatermarkPolicy.limitingRealTimeLag()`. This
+policy emits a watermark that lags behind the system clock by a fixed
+amount. This policy doesn't depend on the events processed, it
+depends solely on the system clock. Currently, this policy is available
+only in Core API.
```

Comment:
In newer versions (since the website) we no longer have any documentation on `WatermarkPolicy`, so  for non-experts it might not be obvious how they can make use of it. Maybe, at least, add some links to javadoc here or explain a bit more?

Reply:
This is a design document for us and for expert users...
Documentation for normal users is in [https://github.com/hazelcast/hazelcast-jet/blob/0cf6b76368fff994348554a80000b9c9573bd5fa/hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/StreamSourceStage.java#L47-L71](Javadoc).
And for non-experts stumbling over this doc some exposure to internals is good...